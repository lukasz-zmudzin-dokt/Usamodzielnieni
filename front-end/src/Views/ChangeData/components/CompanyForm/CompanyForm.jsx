import React from "react";
import { Card } from "react-bootstrap";
import { FormGroup } from "components";

const CompanyForm = ({ setData, data }) => {
  const { company_address, company_name } = data;
  return (
    <Card bg="light" className="changeData__wrapper__card">
      <Card.Header>Adres firmy</Card.Header>
      <Card.Body>
        <FormGroup
          header="Nazwa firmy"
          setVal={(val) => setData({ ...data, company_name: val })}
          val={company_name}
          length={{ min: 1, max: 100 }}
          id="companyName"
          required
          incorrect="Podaj poprawną nazwę firmy."
        />
        <FormGroup
          header="Ulica"
          required
          setVal={(val) =>
            setData({
              ...data,
              company_address: { ...company_address, street: val },
            })
          }
          val={company_address.street}
          length={{ min: 1, max: 120 }}
          id="streetC"
          incorrect="Podaj poprawną nazwę ulicy."
        />
        <FormGroup
          header="Numer budynku"
          required
          setVal={(val) =>
            setData({
              ...data,
              company_address: {
                ...company_address,
                street_number: val,
              },
            })
          }
          pattern="^([0-9]{1})[\s\S]*"
          val={company_address.street_number}
          length={{ min: 1, max: 20 }}
          id="streetNumberC"
          incorrect="Podaj poprawny numer budynku."
        />
        <FormGroup
          header="Nazwa miasta"
          required
          setVal={(val) =>
            setData({
              ...data,
              company_address: { ...company_address, city: val },
            })
          }
          val={company_address.city}
          length={{ min: 1, max: 40 }}
          id="cityC"
          incorrect="Podaj poprawną nazwę miasta."
        />
        <FormGroup
          header="Kod pocztowy"
          required
          setVal={(val) =>
            setData({
              ...data,
              company_address: {
                ...company_address,
                postal_code: val,
              },
            })
          }
          val={company_address.postal_code}
          length={{ min: 1, max: 6 }}
          pattern="[0-9]{2}[-][0-9]{3}"
          incorrect="Podaj poprawny kod pocztowy."
          id="postalCodeC"
        />
      </Card.Body>
    </Card>
  );
};

export default CompanyForm;

// <Form.Group controlId="formGroupCompanyNip" className="">
//     <Form.Control
//         name="company_nip"
//         type="text"
//         pattern={"[0-9]{9}" + this.getNipControlNum()}
//         placeholder="NIP"
//         onChange={(e) => onChange(onBlur, data, e)}
//         required
//         minLength="10"
//         maxLength="10"
//     />
//     <Form.Control.Feedback type="invalid">
//         Podaj prawidłowy NIP
//             </Form.Control.Feedback>
// </Form.Group>
