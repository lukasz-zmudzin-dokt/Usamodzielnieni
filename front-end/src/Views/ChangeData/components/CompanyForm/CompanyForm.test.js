import React from "react";
import { render } from "@testing-library/react";
import CompanyForm from "./CompanyForm";

describe("CompanyForm", () => {
  let data = {
    first_name: "user",
    last_name: "user",
    phone_number: "+48123123123",
    company_name: "abc",
    company_address: {
      city: "new",
      street: "city",
      street_number: "12",
      postal_code: "12-123",
    },
    nip: "123123123",
  };

  it("should match snapshot", () => {
    const { container } = render(
      <CompanyForm data={data} setData={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
