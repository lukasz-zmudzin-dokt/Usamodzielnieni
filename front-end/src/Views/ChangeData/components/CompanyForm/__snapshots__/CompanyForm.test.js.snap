// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyForm should match snapshot 1`] = `
<div>
  <div
    class="changeData__wrapper__card card bg-light"
  >
    <div
      class="card-header"
    >
      Adres firmy
    </div>
    <div
      class="card-body"
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="companyName"
        >
          Nazwa firmy
        </label>
        <input
          class="form-control"
          data-testid="default"
          id="companyName"
          maxlength="100"
          minlength="1"
          placeholder="Nazwa firmy"
          required=""
          type="text"
          value="abc"
        />
        <div
          class="invalid-feedback"
        >
          Podaj poprawną nazwę firmy.
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="streetC"
        >
          Ulica
        </label>
        <input
          class="form-control"
          data-testid="default"
          id="streetC"
          maxlength="120"
          minlength="1"
          placeholder="Ulica"
          required=""
          type="text"
          value="city"
        />
        <div
          class="invalid-feedback"
        >
          Podaj poprawną nazwę ulicy.
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="streetNumberC"
        >
          Numer budynku
        </label>
        <input
          class="form-control"
          data-testid="default"
          id="streetNumberC"
          maxlength="20"
          minlength="1"
          pattern="^([0-9]{1})[\\\\s\\\\S]*"
          placeholder="Numer budynku"
          required=""
          type="text"
          value="12"
        />
        <div
          class="invalid-feedback"
        >
          Podaj poprawny numer budynku.
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="cityC"
        >
          Nazwa miasta
        </label>
        <input
          class="form-control"
          data-testid="default"
          id="cityC"
          maxlength="40"
          minlength="1"
          placeholder="Nazwa miasta"
          required=""
          type="text"
          value="new"
        />
        <div
          class="invalid-feedback"
        >
          Podaj poprawną nazwę miasta.
        </div>
      </div>
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="postalCodeC"
        >
          Kod pocztowy
        </label>
        <input
          class="form-control"
          data-testid="default"
          id="postalCodeC"
          maxlength="6"
          minlength="1"
          pattern="[0-9]{2}[-][0-9]{3}"
          placeholder="Kod pocztowy"
          required=""
          type="text"
          value="12-123"
        />
        <div
          class="invalid-feedback"
        >
          Podaj poprawny kod pocztowy.
        </div>
      </div>
    </div>
  </div>
</div>
`;
